File: js/src/types/record.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordVersion = exports.RECORD_V1_SIZE = exports.Record = void 0;
/**
 * List of SNS Records
 */
var Record;
(function (Record) {
    Record["IPFS"] = "IPFS";
    Record["ARWV"] = "ARWV";
    Record["SOL"] = "SOL";
    Record["ETH"] = "ETH";
    Record["BTC"] = "BTC";
    Record["LTC"] = "LTC";
    Record["DOGE"] = "DOGE";
    Record["Email"] = "email";
    Record["Url"] = "url";
    Record["Discord"] = "discord";
    Record["Github"] = "github";
    Record["Reddit"] = "reddit";
    Record["Twitter"] = "twitter";
    Record["Telegram"] = "telegram";
    Record["Pic"] = "pic";
    Record["SHDW"] = "SHDW";
    Record["POINT"] = "POINT";
    Record["BSC"] = "BSC";
    Record["Injective"] = "INJ";
    Record["Backpack"] = "backpack";
    Record["A"] = "A";
    Record["AAAA"] = "AAAA";
    Record["CNAME"] = "CNAME";
    Record["TXT"] = "TXT";
    Record["Background"] = "background";
    Record["BASE"] = "BASE";
    Record["IPNS"] = "IPNS";
})(Record || (exports.Record = Record = {}));
exports.RECORD_V1_SIZE = new Map([
    [Record.SOL, 96],
    [Record.ETH, 20],
    [Record.BSC, 20],
    [Record.Injective, 20],
    [Record.A, 4],
    [Record.AAAA, 16],
    [Record.Background, 32],
]);
var RecordVersion;
(function (RecordVersion) {
    RecordVersion[RecordVersion["V1"] = 1] = "V1";
    RecordVersion[RecordVersion["V2"] = 2] = "V2";
})(RecordVersion || (exports.RecordVersion = RecordVersion = {}));

File: python/types/record.py
from enum import Enum


class Record(Enum):
    IPFS = "IPFS"
    ARWV = "ARWV"
    SOL = "SOL"
    ETH = "ETH"
    BTC = "BTC"
    LTC = "LTC"
    DOGE = "DOGE"
    Email = "email"
    Url = "url"
    Discord = "discord"
    Github = "github"
    Reddit = "reddit"
    Twitter = "twitter"
    Telegram = "telegram"
    Pic = "pic"
    SHDW = "SHDW"
    POINT = "POINT"
    BSC = "BSC"
    Injective = "INJ"
    Backpack = "backpack"
    A = "A"
    AAAA = "AAAA"
    CNAME = "CNAME"
    TXT = "TXT"
    Background = "background"
    BASE = "BASE"
    IPNS = "IPNS"


class RecordVersion(Enum):
    V1 = 1
    V2 = 2


RECORD_V1_SIZE = {
    Record.SOL: 96,
    Record.ETH: 20,
    Record.BSC: 20,
    Record.Injective: 20,
    Record.A: 4,
    Record.AAAA: 16,
    Record.Background: 32,
}